src/Server/RestRouterHandlers/ColumnDefinition.cpp

```diff
@@ -42,38 +55,48 @@ String getCreateColumnDefination(const Poco::JSON::Object::Ptr & column)
     return boost::algorithm::join(column_definition, " ");
 }
 
-String getUpdateColumnDefination(const Poco::JSON::Object::Ptr & payload, String & column_name)
+String getUpdateColumnDefination(const Poco::JSON::Object::Ptr & payload, const String & database, const String & table, String & column)
 {
     std::vector<String> update_segments;
     if (payload->has("name"))
     {
-        update_segments.push_back(" RENAME COLUMN " + column_name + " TO " + payload->get("name").toString());
-        column_name = payload->get("name").toString();
+        update_segments.push_back(" RENAME COLUMN IF EXISTS " + column + " TO " + payload->get("name").toString());
+        column = payload->get("name").toString();
     }
 
-    if (payload->has("comment"))
+    if (payload->has("type"))
     {
-        update_segments.push_back(" COMMENT COLUMN " + column_name + " COMMENT " + payload->get("comment").toString());
+        update_segments.push_back(" MODIFY COLUMN IF EXISTS " + column + " " + payload->get("type").toString());
     }
 
-    if (payload->has("type"))
+    if (payload->has("default"))
     {
-        update_segments.push_back(" MODIFY COLUMN " + column_name + " " + payload->get("type").toString());
+        const auto & catalog_service = CatalogService::instance(nullptr);
+        const auto & type = catalog_service.getColumnType(database, table, column);
+
+        String default_str = payload->get("default").toString();
+
+        if (type == "String")
+        {
+            default_str = "'" + default_str + "'";
+        }
+
+        update_segments.push_back(" MODIFY COLUMN IF EXISTS " + column + " DEFAULT " + default_str);
     }
 
-    if (payload->has("default"))
+    if (payload->has("comment"))
     {
-        update_segments.push_back(" MODIFY COLUMN " + column_name + " DEFAULT " + payload->get("default").toString());
+        update_segments.push_back(" COMMENT COLUMN IF EXISTS " + column + "'" + payload->get("comment").toString() + "'");
     }
 
     if (payload->has("ttl_expression"))
     {
-        update_segments.push_back(" MODIFY COLUMN " + column_name + " TTL " + payload->get("ttl_expression").toString());
+        update_segments.push_back(" MODIFY COLUMN IF EXISTS " + column + " TTL " + payload->get("ttl_expression").toString());
```

Comment:
We shall remove `IF EXISTS` since it hides issues ?

Reply:
fixed